src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -436,7 +436,8 @@ public override void Configure(IApplicationBuilder app)
         }
         else
         {
-            // Force redirect_url to be https, when authentication on BarViz
+            // Force redirect_url to be https, when authentication on BarViz.
+            // Otherwise it deafults to http, since that's how we're talking to Application Gateway
```

Comment:
Sorry about this but I would really invest some energy into this - it is still not clear why we suddenly want this for one of the endpoints, why does it not apply to other etc...

I think this is really important because 1 year from now someone will be looking at this startup (maybe when moving the service onto container apps?) and scratching their head.

Also there's 2 typos now

```suggestion
            // Force redirect_url to be https, when authenticating on BarViz.
            // Otherwise it defaults to http, since that's how we're talking to Application Gateway
```

Reply:
I suggest to:
1. Better describe the App Gateway - we're not talking to the gateway over HTTP, that's misleading. Please fix the description - it's the other way around.
2. Include that this is the redirect URL we are giving to GitHub and it's sending the token back to us so we need to be on HTTPS

Reply:
We're actually not talking to GitHub via HTTP, we are doing that over HTTPS, so the token should be encrypted. The only thing that is currently HTTP, is the URL that's given as the redirect_url.
I'll improve the comments

Reply:
> We're actually not talking to GitHub via HTTP, we are doing that over HTTPS

But GitHub talks back by redirecting to a non-HTTPS URL. So GH is talking to us, giving us the token back, over HTTP

Reply:
Or you mean the gateway than forces HTTPS?

Reply:
I'm not sure if GitHub is giving us the token back in the same call it's redirecting us on. But setting everything to HTTPS should keep us safe for sure

Reply:
There's more than just one call exchanged with GitHub https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps

Reply:
Where? I don't see it? It's just a normal OAuth flow:

![image](https://github.com/dotnet/arcade-services/assets/7013027/74b1a779-7d5d-4931-9d13-72f7c25b7cda)

1. There's a call to GH
2. GH redirects back
3. We then do whatever with the token

There's no more calls?

Reply:
The way I understand it:
- We make a call to `https://github.com/login/oauth/authorize`, login, and get a temporary `code`
- We make a call to `https://github.com/login/oauth/access_token` exchanging the temporary `code` for the token
- We're redirected

Reply:
So the problem is that GH is redirecting to `HTTP://`. That's the crux.

Reply:
yup, that's how I understand it too.
When I'm inspecting the browser network, I only see one call tho, and with this change, the reditect_uri is https

Reply:
Great. Can you then change the comment to better explain what's going on? That it modifies the URL we're giving

I suggest to:

- Better describe the App Gateway - we're not talking to the gateway over HTTP, that's misleading. Please fix the description - it's the other way around - App Gateway talks to Maestro and that's why it's generating the HTTP redirect URL instead.
- Include that this is the redirect URL we are giving to GitHub and it's sending the token back to us so we need to be on HTTPS